### PR TITLE
Added ability to transfer components

### PR DESCRIPTION
**Pull request by prophetsfall**
This introduces the ability for one character to transfer comps to another. 
It also adds expiration to the recipe table

### CODE REVIEW

File: app/models/ingredient_transfer.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class IngredientTransfer < ApplicationRecord
+belongs_to :source_inventory, class_name: 'Inventory', foreign_key: :source_inventory_id
+belongs_to :target_inventory, class_name: 'Inventory', foreign_key: :target_inventory_id
+belongs_to :ingredient, class_name: 'Ingredient', foreign_key: :ingredient_id
+
+has_many :inventories
+
+end
```
Comment (enthusiastick) [inline app/models/ingredient_transfer.rb]: Your continual disregard for standardized indentation (as outlined in the Ruby style guide) makes me sad in my soul.

File: app/models/recipe.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-class Recipe < ApplicationRecord
+ class Recipe < ApplicationRecord
```
Comment (enthusiastick) [inline app/models/recipe.rb]: … really

Comment (enthusiastick) [inline app/models/recipe.rb]: … really

File: app/controllers/ingredient_transfers_controller.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+   @source_ingredients<<  ingredient.ingredient
+  end
+  @characters = Character.where(archived:false)
+  @ingredient_transfer = IngredientTransfer.new
+end
+
+def create
+  source_character = character = Character.find_by(slug: params[:character_id])
+  ingredient = Ingredient.find_by_id(params["ingredient_transfer"]["ingredient"].to_i)
+  source_inventory = Inventory.where("character_id = ? AND ingredient_id = ?", source_character.id, ingredient.id).first
```
Comment (enthusiastick) [inline app/controllers/ingredient_transfers_controller.rb]: You can specify multiple where conditions in a hash, e.g. `Inventory.where(character: source_character, ingredient: ingredient).first` &ndash; which is equivalent to `Inventory.find_by(character: source_character, ingredient: ingredient)`.

File: app/models/ingredient_transfer.rb (hunk 1 above)
Comment (prophetsfall) [reply app/models/ingredient_transfer.rb]: I'll go ahead and review that, because I do care about your soul

File: app/models/recipe.rb (hunk 2 above)
Comment (prophetsfall) [reply app/models/recipe.rb]: sure wasn't intentional

**Pull request merged**